ballerina/tests/22_variables.bal

```diff
@@ -0,0 +1,220 @@
+    json variables = { "profileId": 1 };
+    string url = "http://localhost:9091/records";
+    json actualPayload = check getJsonPayloadFromBadRequest(url, document, variables);
+    json expectedPayload = check getJsonContentFromFile("fragments_with_undefined_variables.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["variables", "fragments", "input"]
+}
+isolated function testDuplicateVariablesWithMultipleOperations() returns error? {
+    string document = check getGraphQLDocumentFromFile("multiple_operations_with_duplicate_variables.txt");
+    json variables = { "profileId": 1 };
+    string url = "http://localhost:9091/records";
+    json actualPayload = check getJsonPayloadFromBadRequest(url, document, variables, "B");
+    json expectedPayload = check getJsonContentFromFile("multiple_operations_with_duplicate_variables.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["variables", "fragments", "input"]
+}
+isolated function testFragmnetsWithUnsusedVariables() returns error? {
+    string document = check getGraphQLDocumentFromFile("fragments_with_unused_variables.txt");
+    json variables = { "profileId": 1 };
+    string url = "http://localhost:9091/records";
+    json actualPayload = check getJsonPayloadFromBadRequest(url, document, variables, "B");
+    json expectedPayload = check getJsonContentFromFile("fragments_with_unused_variables.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["variables", "fragments", "input"]
+}
+isolated function testInlineFragmentsWithVariables() returns error? {
+    string document = check getGraphQLDocumentFromFile("inline_fragments_with_variables.txt");
+    json variables = { "profileId": 1 };
+    string url = "http://localhost:9091/records";
+    json actualPayload = check getJsonPayloadFromService(url, document, variables);
+    json expectedPayload = check getJsonContentFromFile("inline_fragments_with_variables.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["variables", "fragments", "input"]
+}
+isolated function testVariablesWithDefaultValues() returns error? {
+    string document = check getGraphQLDocumentFromFile("variables_with_default_values.txt");
+    string url = "http://localhost:9091/records";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = check getJsonContentFromFile("variables_with_default_values.json");
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["variables", "inputs", "input_coerce"]
+}
+isolated function testVariablesWithCoerceIntInputToFloat() returns error? {
+    string document = "($weight:float){ weightInPounds(weightInKg:$weight) }";
+    json variables = { "weight": 1};
```

Comment:
When defining a JSON map in Ballerina, you don't necessarily have to use double quotes for the keys.

```suggestion
    json variables = { weight: 1};
```